editor/d2l-rubric-structure-editor.html

```
@@ -344,8 +344,10 @@
 			_handleSaveError: function(e) {
 				this._clearAlerts();
 				e.stopPropagation();
-				this.fire('iron-announce', { text: this.localize('rubricSavingErrorAriaAlert') }, { bubbles: true });
-				this._addAlert('error', e.detail.error.message, this.localize('rubricSavingErrorMessage'));
+				if (e.target === this) {
```

Not clear why this is needed? Adding `e.target === this` breaks some kinds of alerts from child components? What scenario is it breaking without it?

---

without it validation errors throw alerts, which they're not supposed to.

---

Under what scenario? e.stopPropagation is supposed to prevent the error bubbling? Is that not working in some browsers?

---

Actually maybe I'm not understanding what you are saying - let's chat after i'm done this meeting.

---

how are we distinguishing validation errors from other errors currently? What kind of validation errors?  Sounds like we should be preventing those validation errors from bubbling, rather than suppressing errors in this component?  This `_handleSaveError` is supposed to show alerts for errors that are not handled elsewhere e.g. if we get an API error trying to add a criterion group?

---

Ok - Iooking at it again, the reason why these errors are getting this far is because I added the `e.stopPropagation` in the wrong place in `d2l-rubric-criteria-group-editor`. We have code in there which is skipping the showing of the alert at the criteria group level if the error is a 'validation' error. So the call to `e.stopPropagation` should just be moved to the top of that method, then errors won't get here. This was supposed to be a catch all error handler for errors that are not handled more specifically in lower level components. By restricting it to check for `this` it breaks that feature.

But stepping back a little, I kind of think we should stop showing alerts at all for `d2l-siren-entity-error`. That was our original error handling strategy for things like adding criterion and adding criteria groups where we didn't have a good way to show a more specific error bubble.  I think at that stage performSirenAction wasn't even returning a promise.

If we have specific calls to performSirenAction where we don't already have our own custom error handling, then maybe we should raise a specific catch all rubric save error in those locations that can bubble up and cause an alert to be shown. For any instance where we actually highlight a field as being in error it seems superfluous to show the error alert too. Let's discuss with Joseph.

That would simplify things as we wouldn't then have to be trying to suppress `d2l-siren-entity-error` in cases where we actually handle the error and show an error bubble.